### PR TITLE
Workflow - Publish components to Crates.io

### PR DESCRIPTION
**Pull request by aravindgee**
Release automation to crates.io for components chainhook-event-observer and clarity-jupyter.

Related Issue: #557

### CODE REVIEW

File: .github/workflows/release-chainhook-event-observer.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+
+      - name: Publish to Crates
+        run: cargo publish --token ${CRATES_TOKEN}
+        env:
+          CRATES_TOKEN: ${{ secrets.CRATES_TOKEN }}
+        working-directory: ./components/chainhook-event-observer/
```
Comment (lgalabru) [inline .github/workflows/release-chainhook-event-observer.yaml:48]: When you navigate to this crate both a binary/executable and a library can be produced. In this case, we want to produce a library. Should we pass an explicit `--lib` or something?

**Comment by lgalabru** [comment]
In the same vein, could we publish the library [stacks-rpc-client](https://github.com/hirosystems/clarinet/tree/develop/components/stacks-rpc-client) on crates.io ?
Thanks!!

File: .github/workflows/release-chainhook-event-observer.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  pull_request:
+  push:
+    tags:
+     - 'v*'
+    paths-ignore:
+     - '**/CHANGELOG.md'
+     - '**/package.json'
+     - '**/package-lock.json'
+    paths:
+     - components/chainhook-event-observer
```
Comment (lgalabru) [inline .github/workflows/release-chainhook-event-observer.yaml:12]: Same feedback here, we'll need to add a few additional paths:
- components/clarity-repl
- components/clarity-lsp
- components/clarinet-files
- components/clarinet-deployments
- components/chainhook-types-rs

File: .github/workflows/release-clarity-jupyter-kernel.yaml [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  pull_request:
+  push:
+    tags:
+     - 'v*'
+    paths-ignore:
+     - '**/CHANGELOG.md'
+     - '**/package.json'
+     - '**/package-lock.json'
+    paths:
+     - components/clarity-jupyter-kernel
```
Comment (lgalabru) [inline .github/workflows/release-clarity-jupyter-kernel.yaml:12]: Additional paths:
- components/clarity-repl

**Review by lgalabru — COMMENTED** [review]
Thank you for your work on this @aravindgee, just a few tweaks will be required :)

**Comment by aravindgee** [comment]
Release workflow now comprises of these components:
- `components/chainhooks-event-observer`
- `components/stacks-rpc-client`
- `components/clarity-repl`
- `components/clarinet-files`
- `components/clarinet-deployments`
- `components/chainhook-types-rs`
- `components/clarity-jupyter-kernel`


`chainhooks-event-observer` builds as a library only. Does this behavior apply to any of the other components?

**Comment by lgalabru** [comment]
> Does this behavior apply to any of the other components?

Yep, `components/stacks-rpc-client`!

**Comment by smcclellan** [comment]
@aravindgee What's next here?

**Comment by hugocaillard** [comment]
Can be close in favor of #1078

**Pull request closed without merging**